cdap-gateway/src/main/java/co/cask/cdap/gateway/router/RouterAuditLookUp.java

```diff
@@ -145,23 +151,28 @@ private HttpMethod getHttpMethod(Method method) {
     return null;
   }
 
-  private List<Class<?>> getAllHandlerClasses() throws IOException {
+  private List<ClassPath.ClassInfo> getAllHandlerClasses() throws IOException {
     ClassLoader cl = getClass().getClassLoader();
     Map<String, Boolean> cache = new HashMap<>();
     Function<String, URL> lookup = ClassLoaders.createClassResourceLookup(cl);
     ClassPath cp = ClassPath.from(cl);
-    List<Class<?>> results = new ArrayList<>();
+    List<ClassPath.ClassInfo> results = new ArrayList<>();
     for (ClassPath.ClassInfo info : cp.getAllClasses()) {
       if (!info.getPackageName().startsWith("co.cask.cdap")) {
         continue;
       }
       if (Classes.isSubTypeOf(info.getName(), HttpHandler.class.getName(), lookup, cache)) {
-        results.add(info.load());
+        results.add(info);
       }
     }
     return results;
   }
 
+  private boolean isTestClass(ClassPath.ClassInfo classInfo) {
+    URL url = classInfo.url();
+    return url != null && url.getPath().contains("target/test-classes");
```

Comment:
Is this url path correct, even when running from IDE (not `mvn`)?
It places the classes into the `target` directory even then?